enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,252 @@
+if the host name is part of a subdomain configured for HSTS, and false if not.  This may also require
+a new field in `BackendConfig` to hold a list of subdomains to be configured for HSTS.  Details are TBD.
+
+#### Disable HSTS per-route when it is enabled globally
+As mentioned previously, the `Strict-Transport-Security` annotation called 
+`haproxy.router.openshift.io/hsts_header` has a required directive `max-age` that determines
+its lifespan.  A per-route override on globally enabled HSTS can be indicated by setting the `max-age` 
+value in the route's annotation to zero, as shown in this example:
+```
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=0     
+````
+
+### User Stories
+
+#### As a cluster administrator, I want to enable HSTS for all routes globally
+Add an environment variable to the IngressController's deployment spec, for example, setting 
+`ROUTER_DEFAULT_HSTS` to `max-age=31536000` 
+#### As a cluster administrator, I want to enable HSTS on a node-type basis (infra-only, worker-only, or both)
+(TBD - waiting for clarification on this request.  Routes aren't associated with node types.)    
+#### As a cluster administrator, I want to enable HSTS for specific subdomains 
+(TBD - dependency on wildcard DNS record/certificate, or just the `includeSubDomains`)
+#### As a cluster administrator, I want to disable HSTS per-route when it is enabled globally
+Add a route annotation to the route, setting the directive `max-age=0`, e.g.:
+````
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=0     
+````
+### Implementation Details/Notes/Constraints
+Implementing this enhancement requires changes in the following repositories:
+
+- openshift/router
+- openshift/cluster-ingress-operator (TBD)
+
+### Risks and Mitigations
+As previously mentioned, use of the `includeSubDomains` directive may cause problems unless the user
+is aware of its encompassing implications.  To mitigate the risk of unexpected side effects, an analysis 
+of the behavior of `includeSubDomains` will be performed and documented.
+
+Additionally, the direction of overrides on global versus per-route HSTS annotations will be documented,
+because it may not be clear to the user that a global setting cannot override a per-route setting.  Otherwise
+it would not be possible to use a per-route setting.
+
+## Design Details
+
+### Open Questions
+There are numerous TBD sections in this proposal.  It is currently a work in progress until further
+clarification is received.
+
+### Test Plan
+HAProxy and Router have unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional four goals.
+
+The operator has end-to-end tests; for this enhancement, add the following tests:
+
+1. Create an IngressController that enables global HSTS
+2. Open a connection to this route and send a request.
+3. Verify that a response is received and that the headers include HSTS
```

Comment:
Perhaps an e2e test for this enhancement could also verify that HSTS can be disabled on a per route basis using an annotation as you mentioned.